File: guava-cc/src/main/java/com/rainpoetry/common/utils/Logging.java
package com.rainpoetry.common.utils;

/*
 * User: chenchong
 * Date: 2019/3/14
 * description:
 */

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public abstract class Logging {

	protected final String loggerName = getClass().getName();

	protected final Logger logger = LoggerFactory.getLogger(loggerName);

	protected String logIdent;

	protected String msgWithLogIdent(String msg) {
		return logIdent == null ? msg : logIdent + msg;
	}

	protected void info(String msg) {
		logger.info(msgWithLogIdent(msg));
	}

	protected void info(String msg, Throwable t) {
		logger.info(msgWithLogIdent(msg), t);
	}

	protected void debug(String msg) {
		logger.debug(msgWithLogIdent(msg));
	}

	protected void debug(String msg, Throwable t) {
		logger.debug(msgWithLogIdent(msg), t);
	}

	protected void warn(String msg) {
		logger.warn(msgWithLogIdent(msg));
	}

	protected void warn(String msg, Throwable t) {
		logger.warn(msgWithLogIdent(msg), t);
	}

	protected void error(String msg) {
		logger.error(msgWithLogIdent(msg));
	}

	protected void error(String msg, Throwable t) {
		logger.error(msgWithLogIdent(msg), t);
	}

}

File: guava-cc/src/main/scala/scala/rainpoetry/common/Logging.scala
package scala.rainpoetry.common

import org.slf4j.LoggerFactory

/*
 * User: chenchong
 * Date: 2019/4/1
 * description:
 */

trait Logging {

  protected lazy val logger = LoggerFactory.getLogger(loggerName)

  protected var logIdent: String = _

  protected def loggerName: String = getClass.getName

  protected def msgWithLogIdent(msg: String): String = {
    if (logIdent == null) msg else logIdent + msg
  }

  def trace(msg: => String): Unit = logger.trace(msgWithLogIdent(msg))

  def trace(msg: => String, e: => Throwable): Unit = logger.trace(msgWithLogIdent(msg),e)

  def isDebugEnabled: Boolean = logger.isDebugEnabled

  def isTraceEnabled: Boolean = logger.isTraceEnabled

  def debug(msg: => String): Unit = logger.debug(msgWithLogIdent(msg))

  def debug(msg: => String, e: => Throwable): Unit = logger.debug(msgWithLogIdent(msg),e)

  def info(msg: => String): Unit = logger.info(msgWithLogIdent(msg))

  def info(msg: => String,e: => Throwable): Unit = logger.info(msgWithLogIdent(msg),e)

  def warn(msg: => String): Unit = logger.warn(msgWithLogIdent(msg))

  def warn(msg: => String, e: => Throwable): Unit = logger.warn(msgWithLogIdent(msg),e)

  def error(msg: => String): Unit = logger.error(msgWithLogIdent(msg))

  def error(msg: => String, e: => Throwable): Unit = logger.error(msgWithLogIdent(msg),e)

}
